airflow/models/dagrun.py

```diff
@@ -137,7 +138,7 @@ class DagRun(Base, LoggingMixin):
     triggered_by = Column(
         Enum(DagRunTriggeredByType, native_enum=False, length=50)
     )  # Airflow component that triggered the run.
-    conf = Column(PickleType)
+    conf = Column(JSON().with_variant(postgresql.JSONB, "postgresql"))
```

Comment:
Do we not need any more handling than these?

Reply:
I think you _might_ need to change this line:

https://github.com/apache/airflow/blob/3c1124e054f9d795b0e9007041653888da9de6a8/airflow/www/forms.py#L131-L132

Reply:
You will also need to add a `newsfragment`

Reply:
fyi: the following is what I had to do for XComs:

PRs:

1. [Remove XCom pickling](https://github.com/apache/airflow/pull/43905)
2. [Migrate pickled data & change XCom value type to JSON](https://github.com/apache/airflow/pull/44166)
3. [Handle XCom value errors in FAB's XCom list view](https://github.com/apache/airflow/pull/44275)
4. [Use ](https://github.com/apache/airflow/pull/44290)JSONB[ type for ](https://github.com/apache/airflow/pull/44290)XCom.value[ column in PostgreSQL](https://github.com/apache/airflow/pull/44290)